docs: terraform injection

thelongshanx opened the pull request:
**What this PR does / why we need it**:
shows how to inject backend variables through an exec 

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

edvald commented on examples/demo-project-start/.gardenignore:
```diff
@@ -0,0 +1,3 @@
+# Add paths here that you would like Garden to ignore when building modules and computing versions,
```
I think these are here by accident, right? We should add these to .gitignore, since this can easily happen again :P

thelongshanx commented on docs/advanced/terraform.md:
````diff
@@ -81,6 +81,42 @@ Each command automatically applies any variables configured on the provider/modu
+}
+```
+
+In the `project.garden.yaml` file for this sample, the exec provider calls an initScript that removes any pre-existing state files so we can have a clean init, then substitutes the s3 bucket name with the environment name in the backend.tf file.
````
I should've opened up the man page for sed - this is actually a way to change a file in-place! (I was working from Anna's code for my example) I'll update the text.

thelongshanx commented on docs/advanced/terraform.md:
````diff
@@ -81,6 +81,42 @@ Each command automatically applies any variables configured on the provider/modu
+    dependencies: [exec]
+```
+
+Now when you deploy a new Terraformed environment, the new backend statefile will know where to go.
````
I've fixed the sed command description and amended the commit to not include those other files (didn't check out from main 🤦).
However, I'm gonna need help figuring out how to parse that command into an array.

thelongshanx commented on examples/demo-project-start/backend/garden.yml:
```diff
@@ -0,0 +1,841 @@
+kind: Module
```
I've removed these files- I accidentally checked this branch out from a different dev branch

thelongshanx commented:
@edvald @Orzelius I've incorporated the requested changes (except the second way to solve) and deleted irrelevant files - would you mind reviewing again please?
